src/store/index-level.ts

```diff
@@ -1,213 +1,284 @@
-      // 'dateCreated\u0000"2023-05-25T18:23:29.425008Z"\u0000bafyreigs3em7lrclhntzhgvkrf75j2muk6e7ypq3lrw3ffgcpyazyw6pry'
-      const key = this.join(propertyName, this.encodeValue(propertyValue), dataId);
-      operations.push({ type: 'put', key, value: dataId });
+    const tenantPartition = await this.db.partition(tenant);
+    const indexPartition = await tenantPartition.partition(INDEX_SUBLEVEL_NAME);
+    await indexPartition.batch(indexOps, options);
+  }
+
+  async delete(tenant: string, key: string, options?: IndexLevelOptions): Promise<void> {
+    const tenantPartition = await this.db.partition(tenant);
+    const indexPartition = await tenantPartition.partition(INDEX_SUBLEVEL_NAME);
+
+    const indexOps: LevelWrapperBatchOperation<string>[] = [];
+    const indexKey = `__${key}__indexes`;
+    const serializedIndexes = await indexPartition.get(indexKey);
+    if (serializedIndexes === undefined) {
+      return;
     }
 
-    // create a reverse lookup entry for data ID -> its indexes
-    // this is for indexes deletion (`delete()`): so that given the data ID, we are able to delete all its indexes
-    // we can consider putting this info in a different data partition if this ever becomes more complex/confusing
-    operations.push({ type: 'put', key: `__${dataId}__indexes`, value: JSON.stringify(indexes) });
+    const { indexes, sortIndexes } = JSON.parse(serializedIndexes);
+    // delete the reverse lookup
+    indexOps.push({ type: 'del', key: indexKey });
 
-    await partition.batch(operations, options);
+    // delete all indexes associated with the data of the given ID
+    for (const propertyName in indexes) {
+      const propertyValue = indexes[propertyName];
+      for (const sortProperty in sortIndexes) {
+        const sortValue = sortIndexes[sortProperty];
+        const sortedKey = this.constructIndexedKey(
+          sortProperty,
+          propertyName,
+          this.encodeValue(propertyValue),
+          this.encodeValue(sortValue),
+          key,
+        );
+        indexOps.push({ type: 'del', key: sortedKey });
+      }
+    }
+
+    await indexPartition.batch(indexOps, options);
   }
 
-  /**
-   * Executes the given single filter query and appends the results without duplicate into `matchedIDs`.
-   */
-  private async executeSingleFilterQuery(tenant: string, filter: Filter, matchedIDs: Set<string>, options?: IndexLevelOptions): Promise<void> {
+  async query(tenant:string, queries: FilteredQuery[], options?: IndexLevelOptions): Promise<T[]> {
+    const matched:Map<string, T> = new Map();
+    await Promise.all(queries.map(query => this.executeSingleFilterQuery(tenant, query, matched, options)));
+    return [...matched.values()];
+  }
+
+  async executeSingleFilterQuery(tenant:string, query: FilteredQuery, matches: Map<string, T>, options?: IndexLevelOptions): Promise<void> {
```

Comment:
Can we add light documentation on what `matches` is for code readers? Specifically on the `string` since `matches` acts as return type which is kinda important.